ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -76,29 +78,28 @@ def forward(
         sequence_length: int = 1,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         encodes = []
-        for idx, encoder in enumerate(self.vector_encoders):
+
+        for idx, processor in enumerate(self.vector_inputs):
             vec_input = vec_inputs[idx]
```

Comment:
I think the vector obs should be all concatenated with each other before the first linear layers.

Reply:
This leaves open the possibility to have separate processors for different vector obs. The processed vec_obs are all concatenated before the first linear layers.